golos.publication/golos.publication.cpp

```diff
@@ -57,6 +62,37 @@ void publication::create_message(name account, std::string permlink,
     eosio_assert(pool != pools.end(), "publication::create_message: [pools] is empty");
     pool = --pools.end();
     check_account(account, pool->state.funds.symbol);
+    
+    tables::limit_table lims(_self, _self.value);
+    auto lim_itr = lims.find(parentacc ? structures::limitparams::COMM : structures::limitparams::POST);
+    eosio_assert(lim_itr != lims.end(), "publication::create_message: limit parameters not set");
+    auto token_code = pool->state.funds.symbol.code();
+    eosio_assert(golos::vesting::get_account_effective_vesting(config::vesting_name, account, token_code).amount >= lim_itr->min_vesting,
+        "insufficient effective vesting amount");
+    auto issuer = eosio::token::get_issuer(config::token_name, token_code);
+    auto message_id = hash64(permlink);
+    if(lim_itr->price >= 0)
+        INLINE_ACTION_SENDER(charge, use) (config::charge_name, 
+            {issuer, config::invoice_name}, {
+                account, 
+                token_code, 
+                lim_itr->charge_id, 
+                lim_itr->price, 
+                lim_itr->cutoff,
+                vestpayment ? lim_itr->vesting_price : 0
```

Comment:
It is not convenient to store this data in a charge contract, because several actions with different cutoffs and prices can be tied to one charge. Theoretically one charge can serve actions of several contracts with its own price logic (In a case of voting, for example, the price depends on a weight). I don’t think that all of this should be included in the charge contract.